test/com/facebook/buck/cli/TestRunningTest.java

```diff
@@ -165,6 +165,28 @@ public void testNonGeneratedSourceFile() {
     verify(defaultJavaPackageFinder);
   }
 
+  @Test
+  public void testNonGeneratedSourceFileWithoutPathElements() {
+    Path pathToNonGenFile = Paths.get("package/src/SourceFile1.java");
+    assertFalse(MorePaths.isGeneratedFile(pathToNonGenFile));
+
+    ImmutableSortedSet<Path> javaSrcs = ImmutableSortedSet.of(pathToNonGenFile);
+    JavaLibrary javaLibrary = new FakeJavaLibrary(
+        BuildTarget.builder("//foo", "bar").build(),
+        new SourcePathResolver(new BuildRuleResolver())).setJavaSrcs(javaSrcs);
+
+    DefaultJavaPackageFinder defaultJavaPackageFinder =
+        createMock(DefaultJavaPackageFinder.class);
+    expect(defaultJavaPackageFinder.getPathsFromRoot()).andReturn(pathsFromRoot);
+    expect(defaultJavaPackageFinder.getPathElements()).andReturn(ImmutableSet.<String>of("/"));
+
+    replay(defaultJavaPackageFinder);
+
+    ImmutableSet<String> result = TestRunning.getPathToSourceFolders(
+        javaLibrary, Optional.of(defaultJavaPackageFinder), new FakeProjectFilesystem());
```

Comment:
Hmm, actually, I'm not sure what you wanted to happen here.  That set appears to be empty, which I don't think is the right behavior.